pkg/controllers/startup_policy.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+)
+
+// replicatedJobsAtLeastReady tells if the replicated jobSet is ready
+// or it has finished
+// If replicas is equal then we assume that replicatedJob is considered ready
+func replicatedJobsAtLeastReady(replicas int32, rjJobStatus jobset.ReplicatedJobStatus) bool {
```

Comment:
Sorry, are we doing started or ready? I was thinking ready as that’s is most likely what users would want.

Reply:
It is called startup policy :)

Remember that we are not only counting ready jobs, but completed one too.

Reply:
I don't really understand. Technically we have always started the jobs in order. "Starting" in this case would just mean submitting to the k8sclient and creating the job. If a user wants startup policy, wouldn't they also want the replicated job to be ready? I am thinking of cases where you have a driver replicated job and worker replicated jobs. In that case, having the driver be ready before the workers start would be the main use case I would think.

I am using the following equation for determining ready:

```
func replicatedJobsAtLeastReady(replicas int32, rjJobStatus jobset.ReplicatedJobStatus) bool {
	return replicas == rjJobStatus.Failed+rjJobStatus.Ready+rjJobStatus.Succeeded
}
```

AFAIK ready means that the readiness probe is successful for the pod so that once the pod has a successful readiness probe, the pod is considered ready. I think this makes sense to me as that is aligned with how I could envision someone using startupPolicy. I can't really think of any reason why someone would want a state before ready if they specify a startup policy.

I don't fully understand what we mean by started in this case.

Reply:
I am not asking to change the equation, I am only asking to change the function name.

From my perspective, started = atLeastReady; started (past verb) means the job was scheduled and started to run at some point in time in the past, it may still be running (jJobStatus.Ready) or completed (rjJobStatus.Failed+rjJobStatus.Succeeded)

> AFAIK ready means that the readiness probe is successful for the pod so that once the pod has a successful readiness probe, the pod is considered ready. 

Yes, I am not disagreeing with this.

Reply:
I see. Sorry I wasn't sure what you meant by started. We are on the same page then. I will be change the name soon.